src/fs/dir.rs

```diff
@@ -48,7 +48,9 @@ impl Dir {
     /// [`std::fs::File::open`]: https://doc.rust-lang.org/std/fs/struct.File.html#method.open
     #[inline]
     pub fn open_file<P: AsRef<Path>>(&self, path: P) -> io::Result<File> {
-        self.open_file_with(path, OpenOptions::new().read(true))
+        // We're delegating to a sys-dependent open_file due to `openat2` being really strict
+        // about mixing oflags and mode bits.
+        self.sys.open_file(path.as_ref())
```

Comment:
Instead of bubbling this up, could we handle this locally within the `openat2`-specific code? That is, instead of it doing:
```rust
    let mode = options.ext.mode;
```
have it check if the flags contain O_CREAT or O_TMPFILE and set the mode to 0 if not?

Reply:
Done, albeit in a very roundabout fashion since `yanix` crate doesn't expose `O_TMPFILE` yet. This will need to get fixed on the `yanix` side. I left a TODO as a reminder what needs to be done.